### PR TITLE
OpenID Connect Implementation (OIDC)

### PR DESCRIPTION
**Pull request by minozheros**
This PR implements OIDC as discussed in https://github.com/yasudacloud/strapi-plugin-sso/issues/26

### CODE REVIEW

**Review by FlipperLP — APPROVED** [review]

File: server/controllers/oidc.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+const axios = require("axios");
+const { v4 } = require('uuid');
+const { getService } = require("@strapi/admin/server/utils");
+
+const configValidation = () => {
+  const config = strapi.config.get('plugin.strapi-plugin-sso')
+  if (config['OIDC_CLIENT_ID'] && config['OIDC_CLIENT_SECRET'] && config['OIDC_ISSUER']
```
Comment (yasudacloud) [inline server/controllers/oidc.js:7]: `OIDC_ISSUER` does not seem to be used, so I would like to remove it if it is not a problem

**Review by yasudacloud — COMMENTED** [review]
@minozheros 
I appreciate your work on this. Thank you for the implementation.
While the overall operation seems fine, I have a couple of requests:

Could we please consider removing the unused environment variable OIDC_ISSUER?
Additionally, it would be helpful to add a note about OIDC in the README.

I apologize for any inconvenience caused by these requests.

File: server/controllers/oidc.js (hunk 1 above)
Comment (minozheros) [reply server/controllers/oidc.js:7]: Sure go ahead.

File: server/controllers/oidc.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  const config = strapi.config.get('plugin.strapi-plugin-sso')
+  if (config['OIDC_CLIENT_ID'] && config['OIDC_CLIENT_SECRET'] && config['OIDC_ISSUER'] 
+      && config['OIDC_REDIRECT_URI'] && config['OIDC_SCOPES']
+      && config['OIDC_TOKEN_ENDPOINT'] && config['OIDC_USER_INFO_ENDPOINT'] 
+      && config['OIDC_GRANT_TYPE'] && config['OIDC_FAMILY_NAME_FIELD'] 
+      && config['OIDC_GIVEN_NAME_FIELD'] && config['OIDC_AUTHORIZATION_ENDPOINT']
+      ) {
+    return config
+  }
+  throw new Error('OIDC_AUTHORIZATION_ENDPOINT,OIDC_TOKEN_ENDPOINT, OIDC_USER_INFO_ENDPOINT,OIDC_CLIENT_ID, OIDC_CLIENT_SECRET, OIDC_ISSUER, OIDC_REDIRECT_URI, and OIDC_SCOPES are required')
```
Comment (yasudacloud) [inline server/controllers/oidc.js:15]: @minozheros 
There was one more `OIDC_ISSUER` location left!
Sorry.

**Pull request merged**